sdcm/provision/azure/utils.py

```diff
@@ -67,8 +67,9 @@ def get_scylla_images(  # pylint: disable=too-many-branches
                     break
             else:
                 output.append(image)
+    output.sort(key=lambda img: int(img.tags.get('build_id', "0")))
+    output.sort(key=lambda img: int(SCYLLA_VERSION_GROUPED_RE.match(img.tags.get("ScyllaVersion")).group("date")))
```

Comment:
It will fail because `match` returns `None` when there are no results. Hence, you will get an `AttributeError` exception trying to call `group()` method from `None` object.

Reply:
Not exactly, it won't be executed at all when `output` is empty.
Anyway, it will fail unexpectedly when there's no `ScyllaVersion` or `build_id` tags.
I'll try to add some unit tests to verify this version regex and possible outputs from Azure service.

Reply:
It is not related to the `empty output` case, If `match` doesn't happen (which is totally possible) then we get unhandled exception as mentioned before.

Reply:
yep, I've added handling of that situation. Thanks.